s3path.py

```diff
@@ -754,20 +755,20 @@ class PureS3Path(PurePath):
     __slots__ = ()
 
     @classmethod
-    def from_uri(cls, uri):
+    def from_uri(cls, uri: str):
```

Comment:
I'm curious to understand why you didn't type hint the output of this method and `from_bucket_key`. Naively I would have thought you could do
```python
@classmethod
def from_uri(cls, uri: str) -> "PureS3Path":
```
Is there some reason not to do this?

Reply:
I'm just splitting the change in multiple PRs to avoid doing a big one,it's possible to do this indeed